docs/plugins.md

```diff
@@ -635,3 +635,65 @@ We avoid calling `valid()` on the first iteration of the inner loop by using `do
 position on the outer `while`. Using an inner `while` instead is functionally equivalent, but calling `valid()` twice in
 succession would cause the same documents to be fetched twice from Solr (although still processed once by the script) on common
 multiples of the chunk size and prefetch size.
+
+PostBigExtractRequest plugin
+---------------------
+
+If you use complex Solr extract queries with lots of literals to define your custom metadata the total length of your querystring can exceed the default limits of servlet containers. One solution is to alter your servlet container configuration to raise this limit. But if this is not possible or desired this plugin is another way to solve the problem.
+
+This plugin can automatically move all parameters from querystring to the multipart body content of the request if the querystring exceeds a length limit.
+
+For instance, in Jetty the default ‘headerBufferSize’ is 4KiB. Tomcat 10 has a similar setting ‘maxHttpHeaderSize’, 8KiB by default. This limit applies to all the combined headers of a request, so it’s not just the querystring. In comparison, the default for POST data in tomcat (‘maxPostSize’) is 2MiB. Jetty uses a ‘maxFormContentSize’ setting with a lower default value of 200kB, but still way higher than the header limit and well above the length of even the most complex queries.
+
+The plugin only uses the length of the querystring to determine the parameters relocation. Other headers are not included in the length calculation so your limit should be somewhat lower than the limit of the servlet container to allow for room for other headers. This was done to keep the length calculation simple and fast because the exact headers used can vary for the various client adapters available in Solarium. You can alter the `maxquerystringlength` by using a config setting or the API. Only `Extract` queries are affected, other types of queries are not altered.
+
+### Example usage
+
+```php
+<?php
+
+/*
+ * This file is part of the Solarium package.
+ *
+ * For the full copyright and license information, please view the COPYING
+ * file that was distributed with this source code.
+ */
```

Comment:
We don't include this in the examples. People would just copy/paste it into their own codebase. 😉